### ISSUE TITLE
Add the ability to safely read into ReadBuf

### ISSUE THREAD

**Issue by notgull** [hyperium/hyper#3509]
**Is your feature request related to a problem? Please describe.**
It is impossible to write an implementation of `hyper::rt::Read` because `ReadBufCursor` cannot be written to safely.

**Describe the solution you'd like**
Have a method that initializes the read buffer by zeroing it and then returning an `&mut [u8]` that can be filled safely. E.g. [this](https://docs.rs/tokio/latest/tokio/io/struct.ReadBuf.html#method.initialize_unfilled).

**Describe alternatives you've considered**
N/A

**Comment by seanmonstar** [comment]
Yea, I purposefully stuck with the smallest API possible, so that an overly eager method wouldn't restrict it's internal design. I'm happy to have proposals for new methods to add.

You're specifically asking for a way to access the slice as `&mut [u8]`, instead of the `&mut [MaybeUninit<u8>]`, right?

**Comment by notgull** [comment]
Yes, that would be best. See [my implementation](https://codeberg.org/notgull/smol-hyper/src/commit/192d1116b3b0f28d0729e7c6206c0ed5fadb3bc3/src/rt/futures.rs#L78-L97) where I have to use `unsafe` code to read into the buffer. If I could pass a `&mut [u8]` to the underlying reader I would be able to mark this crate as `forbid(unsafe_code)`.
